.gitignore

```diff
@@ -1,21 +1,69 @@
-# Generated by Cargo
-# will have compiled files and executables
-debug/
-target/
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
 
-# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
-# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+apache-rat-*.jar
+
+# Linux perf sample data
+perf.data
+perf.data.old
+
+
+.vscode
+.idea/
+.pytest_cache/
+pkgs
+docker_cache
+.gdb_history
+*.orig
+.*.swp
+.*.swo
+
+venv/*
+
+# macOS
+.DS_Store
+
+# docker volumes used for caching
+.docker
+
+# Rust
+target
 Cargo.lock
+!datafusion-cli/Cargo.lock
```

Comment:
some of this looks like it may only be relevant for the datafusion project and not this one